org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/instance/type/BundleValidator.java

```diff
@@ -1393,8 +1473,41 @@ private boolean validateSignatureDigSig(List<ValidationMessage> errors, Element
         }
         xml = canon.contains("xml");
       }
+
+      // 3. purpose
+      String purpose = dsig.getPurpose();
+      String purposeDesc = dsig.getPurposeDesc();
+      boolean purposeOk = false;
+      List<Element> types = signature.getChildren("type");
+      for (Element type : types) {
+        String p = null;
+        String system = type.getNamedChildValue("system");
+        String code = type.getNamedChildValue("code"); 
+        if (OIDUtilities.isValidOID(code)) {
+          p = "urn:oid:"+code;
+        } else {
+          p = system+"#"+code;
+        }
+        if (purpose == null) {

```

Comment:
This seems a bit weird to me. How does the spec work here? If the sig doesn't have a purpose, it feels like we should have branched earlier to handle that case.

Reply:
I don't think so. we'd still have to iterate the type element like we're doing, we'd just have to do that twice. So that would be better?